mlflow/sagemaker/__init__.py

```diff
@@ -158,13 +158,16 @@ def deploy(app_name, model_path, bucket=None, image_url=DEFAULT_IMAGE_URL, run_i
     if run_id:
         model_path = _get_model_log_dir(model_path, run_id)
         prefix = os.path.join(run_id, prefix)
-        run_id = _check_compatible(model_path)
+    run_id = _check_compatible(model_path)
```

Comment:
this changes the behavior ? was it incorrect before?

Reply:
@tomasatdatabricks

Reply:
ah sorry, was viewing a subset of changes - this was changing it back to the original logic